typecast vars to ensure the value is an array

jordyrosenbrand opened the pull request:
The method `callMiddleware` expects the fourth parameter to be an array while in the method `queryAndReturnResult` it can be either an array or null.

emteknetnz committed:
Merge branch '4.0' into 4

emteknetnz committed:
Merge branch '4.0' into 4

GuySartorelli committed:
Merge branch '4.0' into 4

Commit:
typecast vars to ensure the value is an array

ChloeSartorelli commented on src/QueryHandler/QueryHandler.php:
```diff
@@ -102,7 +102,7 @@ public function queryAndReturnResult(GraphQLSchema $schema, $query, ?array $vars
         };
 
-        return $this->callMiddleware($schema, $query, $context, $vars, $last);
+        return $this->callMiddleware($schema, $query, $context, (array) $vars, $last);
```
```suggestion
        return $this->callMiddleware($schema, $query, $context, $vars ?? [], $last);
```
This is the syntax we've used elsewhere in core code for this purpose - we should continue following this pattern.

GuySartorelli reviewed (changes requested):
Thanks! Good catch. Just a small change to make.

Can you please also retarget on the `4.0` branch?

jordyrosenbrand committed:
Update src/QueryHandler/QueryHandler.php

Use null coalescing operator with an empty array instead of typecasting into an array

Co-authored-by: Guy Sartorelli <36352093+GuySartorelli@users.noreply.github.com>

GuySartorelli commented:
The travis failure is also reflected on the `4` branch itself so it not related to this PR: https://app.travis-ci.com/github/silverstripe/silverstripe-graphql/jobs/573407968#L1976
